app/code/Magento/Customer/Block/Account/AuthenticationPopup.php

```diff
@@ -24,21 +27,29 @@ class AuthenticationPopup extends \Magento\Framework\View\Element\Template
      */
     private $serializer;
 
+    /**
+     * @var HttpContext
+     */
+    private $httpContext;
+
     /**
      * @param \Magento\Framework\View\Element\Template\Context $context
      * @param array $data
      * @param \Magento\Framework\Serialize\Serializer\Json|null $serializer
+     * @param HttpContext $httpContext
      * @throws \RuntimeException
      */
     public function __construct(
         \Magento\Framework\View\Element\Template\Context $context,
         array $data = [],
-        ?\Magento\Framework\Serialize\Serializer\Json $serializer = null
+        ?\Magento\Framework\Serialize\Serializer\Json $serializer = null,
+        HttpContext $httpContext = null
```

Comment:
Can we update this type so it's explicitly nullable, this is a requirement for PHP 8.4 to not get deprecation errors.
So change `HttpContext` to `?HttpContext` please. Thanks! :)